syntax = "proto2";

package authentication;

message CreateAccountWeb {
  required int64 requestId = 1;
  required string email = 2;
  required string plaintextPassword = 3;
}

message CreateAccountWebSuccess {
  required int64 requestId = 1;
  required int64 userId = 2;
}

message CreateAccountWebFailure {
  required int64 requestId = 1;
  enum Reason {
    ACCOUNT_ALREADY_EXISTS = 0;
    INVALID_PASSWORD = 1;
    INTERNAL_SERVER_ERROR = 2;
  }
  required Reason reason = 2;
}

message AuthenticateWeb {
  required int64 requestId = 1;
  required string email = 2;
  required string plaintextPassword = 3;
}

message AuthenticateWebSuccess {
  required int64 requestId = 1;
  required int64 userId = 2;
  required string creationDate = 3;
}

message AuthenticateWebFailure {
  required int64 requestId = 1;
  enum Reason {
    INVALID_CREDENTIALS = 0;
    INTERNAL_SERVER_ERROR = 1;
  }
  required Reason reason = 2;
}
